cloverage/src/cloverage/instrument.clj

```diff
@@ -517,18 +551,20 @@
             instrumented-form (try
                                 (wrap f-var line-hint form)
                                 (catch Throwable e
-                                  (throw+ e "Couldn't wrap form %s at line %s"
-                                          form line-hint)))]
-        (eval-form filename form line-hint instrumented-form)
+                                  (throw (ex-info "Error instrumenting form"
+                                                  {:filename filename, :line line-hint, :form form}
+                                                  e))))]
+        (try
+          (eval-form filename form line-hint instrumented-form)
+          (catch Throwable e
+            (throw (ex-info "Error evaluating instrumented form" {} e))))
         instrumented-form)
-      ;; if we run into an error instrumenting a form, log it and return/eval the uninstrumented form, so we can
-      ;; continue
+      ;; if we run into an error instrumenting a form or evaluating it, log it and return/eval the uninstrumented
+      ;; form, so we can continue
       (catch Throwable e
-        (log/error "Error instrumenting form"
-                   (ex-info "Error instrumenting form" {:filename filename
-                                                        :line     line-hint
-                                                        :form     form}
-                            e))
+        (log/error (.getMessage e) "\n"
+                   (binding [*print-meta* true]
+                     (with-out-str (pprint/pprint (Throwable->map e)))))
```

Comment:
While trying to work out the root cause of the bug I wanted to fix I made some tweaks to the errors that get logged which makes debugging errors a lot easier. By pretty-printing the map representation of the whole Exception chain it's easier to see the root cause of the failure as well as the various instrumented/macroexpanded forms involved (including metadata)